cocos2d/core/renderer/RendererWebGL.js

```diff
@@ -47,8 +47,34 @@ cc.rendererWebGL = {
             i,
             len;
         var context = ctx || cc._renderContext;
-        for (i = 0, len = locCmds.length; i < len; i++) {
-            locCmds[i].rendering(context);
+        
+        //prepare batching
+        for (i = 0, len = locCmds.length; i< len; ++i) 
+        {
+            var cmd = locCmds[i];
+            if(!cmd._batched && cmd.configureBatch) //may be set to true by processed cmds during this loop 
+            {
+                cmd.configureBatch(locCmds,i);
+            }
+        }
+
+        for (i = 0, len = locCmds.length; i< len; ++i) {
+            var cmd = locCmds[i];
+            if(cmd._batching)
+            {
+                cmd.batchedRendering(context);
```

Comment:
This function should upload all batched commands' data to GPU and invoke the gl rendering function. Right?